File: backend/internal/services/email_service.go
package services

import (
	"bytes"
	"context"
	"encoding/json"
	"fmt"
	"log"
	"math/rand"
	"net/http"
	"strconv"
	"time"

	"github.com/redis/go-redis/v9"
)

// EmailServiceOptimized optimized email service with Redis storage
type EmailServiceOptimized struct {
	resendAPIKey string
	fromEmail    string
	redis        *redis.Client
}

// NewEmailServiceOptimized create optimized email service
func NewEmailServiceOptimized(resendAPIKey, fromEmail string, redis *redis.Client) *EmailServiceOptimized {
	return &EmailServiceOptimized{
		resendAPIKey: resendAPIKey,
		fromEmail:    fromEmail,
		redis:        redis,
	}
}

// SendEmailRequest email sending request
type SendEmailRequest struct {
	From    string   `json:"from"`
	To      []string `json:"to"`
	Subject string   `json:"subject"`
	HTML    string   `json:"html"`
	Text    string   `json:"text,omitempty"`
}

// SendEmailResponse email sending response
type SendEmailResponse struct {
	ID string `json:"id"`
}

// VerificationCodeInfo verification code information
type VerificationCodeInfo struct {
	Email     string    `json:"email"`
	Code      string    `json:"code"`
	Username  string    `json:"username"`
	ExpiresAt time.Time `json:"expires_at"`
	CreatedAt time.Time `json:"created_at"`
	Attempts  int       `json:"attempts"`
}

const (
	// Redis keys
	VERIFICATION_CODE_PREFIX = "verification_code:"
	RATE_LIMIT_PREFIX       = "rate_limit:"
	
	// Limits
	CODE_EXPIRY_MINUTES     = 5
	RATE_LIMIT_MINUTES      = 1
	MAX_ATTEMPTS_PER_EMAIL  = 3
	MAX_SENDS_PER_HOUR      = 5
)

// SendVerificationEmail send verification email with rate limiting
func (e *EmailServiceOptimized) SendVerificationEmail(email, username string) error {
	ctx := context.Background()
	
	// Check rate limiting
	if err := e.checkRateLimit(ctx, email); err != nil {
		return err
	}
	
	// Generate 6-digit verification code
	code := fmt.Sprintf("%06d", rand.Intn(1000000))
	
	// Store verification code in Redis
	codeInfo := VerificationCodeInfo{
		Email:     email,
		Code:      code,
		Username:  username,
		ExpiresAt: time.Now().Add(CODE_EXPIRY_MINUTES * time.Minute),
		CreatedAt: time.Now(),
		Attempts:  0,
	}
	
	codeJSON, err := json.Marshal(codeInfo)
	if err != nil {
		return fmt.Errorf("failed to marshal verification code: %w", err)
	}
	
	// Store in Redis with expiration
	redisKey := VERIFICATION_CODE_PREFIX + email
	err = e.redis.Set(ctx, redisKey, codeJSON, CODE_EXPIRY_MINUTES*time.Minute).Err()
	if err != nil {
		return fmt.Errorf("failed to store verification code in Redis: %w", err)
	}
	
	// Update rate limiting counter
	e.updateRateLimit(ctx, email)
	
	log.Printf("📧 Verification code stored in Redis for %s: %s (expires in %d minutes)", 
		email, code, CODE_EXPIRY_MINUTES)
	
	// Generate email content
	subject := "AIICG Wallpaper - Email Verification Code"
	html := e.generateVerificationHTML(code, username)
	text := e.generateVerificationText(code, username)
	
	// Send email
	if e.resendAPIKey == "" || e.resendAPIKey == "your-resend-api-key-here" {
		// Mock email service for development
		log.Printf("🔧 Mock Email Service (Development Environment)")
		log.Printf("📧 To: %s", email)
		log.Printf("📋 Subject: %s", subject)
		log.Printf("🔑 Verification Code: %s", code)
		log.Printf("⏰ Valid for: %d minutes", CODE_EXPIRY_MINUTES)
		return nil
	}
	
	return e.sendWithResend(email, subject, html, text)
}

// VerifyCode verify verification code with attempt limiting
func (e *EmailServiceOptimized) VerifyCode(email, inputCode string) error {
	ctx := context.Background()
	
	log.Printf("🔍 Verifying code for %s: input=%s", email, inputCode)
	
	// Get verification code from Redis
	redisKey := VERIFICATION_CODE_PREFIX + email
	codeJSON, err := e.redis.Get(ctx, redisKey).Result()
	if err == redis.Nil {
		log.Printf("❌ No verification code found for email: %s", email)
		return fmt.Errorf("verification code not found or expired")
	} else if err != nil {
		return fmt.Errorf("failed to get verification code from Redis: %w", err)
	}
	
	var codeInfo VerificationCodeInfo
	if err := json.Unmarshal([]byte(codeJSON), &codeInfo); err != nil {
		return fmt.Errorf("failed to unmarshal verification code: %w", err)
	}
	
	log.Printf("📋 Found stored code: %s, expires at: %s, attempts: %d", 
		codeInfo.Code, codeInfo.ExpiresAt.Format("15:04:05"), codeInfo.Attempts)
	
	// Check if expired
	if time.Now().After(codeInfo.ExpiresAt) {
		e.redis.Del(ctx, redisKey)
		log.Printf("⏰ Verification code expired for: %s", email)
		return fmt.Errorf("verification code expired")
	}
	
	// Check attempt limit
	if codeInfo.Attempts >= MAX_ATTEMPTS_PER_EMAIL {
		e.redis.Del(ctx, redisKey)
		log.Printf("🚫 Too many attempts for: %s", email)
		return fmt.Errorf("too many verification attempts, please request a new code")
	}
	
	// Check code
	if codeInfo.Code != inputCode {
		// Increment attempt counter
		codeInfo.Attempts++
		updatedJSON, _ := json.Marshal(codeInfo)
		e.redis.Set(ctx, redisKey, updatedJSON, time.Until(codeInfo.ExpiresAt))
		
		log.Printf("❌ Code mismatch. Expected: %s, Got: %s, Attempts: %d/%d", 
			codeInfo.Code, inputCode, codeInfo.Attempts, MAX_ATTEMPTS_PER_EMAIL)
		return fmt.Errorf("invalid verification code (%d/%d attempts)", codeInfo.Attempts, MAX_ATTEMPTS_PER_EMAIL)
	}
	
	// Remove verification code after successful verification
	e.redis.Del(ctx, redisKey)
	log.Printf("✅ Email verification successful: %s", email)
	return nil
}

// checkRateLimit check if email sending rate limit is exceeded
func (e *EmailServiceOptimized) checkRateLimit(ctx context.Context, email string) error {
	rateLimitKey := RATE_LIMIT_PREFIX + email
	
	// Get current count
	countStr, err := e.redis.Get(ctx, rateLimitKey).Result()
	if err == redis.Nil {
		return nil // No previous sends
	} else if err != nil {
		log.Printf("⚠️ Failed to check rate limit: %v", err)
		return nil // Allow on Redis error
	}
	
	count, err := strconv.Atoi(countStr)
	if err != nil {
		return nil // Allow on parse error
	}
	
	if count >= MAX_SENDS_PER_HOUR {
		return fmt.Errorf("rate limit exceeded: maximum %d verification emails per hour", MAX_SENDS_PER_HOUR)
	}
	
	return nil
}

// updateRateLimit update rate limiting counter
func (e *EmailServiceOptimized) updateRateLimit(ctx context.Context, email string) {
	rateLimitKey := RATE_LIMIT_PREFIX + email
	
	// Increment counter with 1 hour expiration
	pipe := e.redis.Pipeline()
	pipe.Incr(ctx, rateLimitKey)
	pipe.Expire(ctx, rateLimitKey, time.Hour)
	_, err := pipe.Exec(ctx)
	
	if err != nil {
		log.Printf("⚠️ Failed to update rate limit: %v", err)
	}
}

// sendWithResend send email using Resend API
func (e *EmailServiceOptimized) sendWithResend(to, subject, html, text string) error {
	log.Printf("📧 Sending email via Resend to: %s", to)
	
	reqBody := SendEmailRequest{
		From:    e.fromEmail,
		To:      []string{to},
		Subject: subject,
		HTML:    html,
		Text:    text,
	}
	
	jsonData, err := json.Marshal(reqBody)
	if err != nil {
		return fmt.Errorf("failed to marshal email request: %w", err)
	}
	
	req, err := http.NewRequest("POST", "https://api.resend.com/emails", bytes.NewBuffer(jsonData))
	if err != nil {
		return fmt.Errorf("failed to create request: %w", err)
	}
	
	req.Header.Set("Authorization", "Bearer "+e.resendAPIKey)
	req.Header.Set("Content-Type", "application/json")
	
	client := &http.Client{Timeout: 30 * time.Second}
	resp, err := client.Do(req)
	if err != nil {
		return fmt.Errorf("failed to send request: %w", err)
	}
	defer resp.Body.Close()
	
	if resp.StatusCode != http.StatusOK {
		var errorResp map[string]interface{}
		json.NewDecoder(resp.Body).Decode(&errorResp)
		log.Printf("❌ Resend API error: %v", errorResp)
		return fmt.Errorf("resend API error: %d - %v", resp.StatusCode, errorResp)
	}
	
	var response SendEmailResponse
	if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
		return fmt.Errorf("failed to decode response: %w", err)
	}
	
	log.Printf("✅ Email sent successfully, ID: %s", response.ID)
	return nil
}

// generateVerificationHTML generate optimized HTML email template
func (e *EmailServiceOptimized) generateVerificationHTML(code, username string) string {
	return fmt.Sprintf(`
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Email Verification - AIICG Wallpaper</title>
    <style>
        body {
            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, 'Helvetica Neue', Arial, sans-serif;
            line-height: 1.6;
            color: #333;
            margin: 0;
            padding: 0;
            background: linear-gradient(135deg, #667eea 0%%, #764ba2 100%%);
        }
        .email-container {
            max-width: 600px;
            margin: 40px auto;
            background-color: #ffffff;
            border-radius: 16px;
            box-shadow: 0 10px 30px rgba(0, 0, 0, 0.1);
            overflow: hidden;
        }
        .header {
            background: linear-gradient(135deg, #667eea 0%%, #764ba2 100%%);
            color: white;
            text-align: center;
            padding: 40px 20px;
        }
        .logo {
            font-size: 32px;
            font-weight: bold;
            margin-bottom: 8px;
        }
        .subtitle {
            font-size: 16px;
            opacity: 0.9;
        }
        .content {
            padding: 40px 30px;
        }
        .greeting {
            font-size: 18px;
            margin-bottom: 20px;
            color: #333;
        }
        .code-section {
            text-align: center;
            margin: 30px 0;
        }
        .code-label {
            font-size: 14px;
            color: #666;
            margin-bottom: 10px;
        }
        .code-box {
            background: linear-gradient(135deg, #667eea 0%%, #764ba2 100%%);
            border-radius: 12px;
            padding: 25px;
            margin: 20px 0;
            display: inline-block;
        }
        .code {
            font-size: 36px;
            font-weight: bold;
            color: #ffffff;
            letter-spacing: 8px;
            font-family: 'Courier New', monospace;
        }
        .instructions {
            background-color: #f8f9fa;
            border-left: 4px solid #667eea;
            padding: 20px;
            margin: 25px 0;
            border-radius: 0 8px 8px 0;
        }
        .warning {
            background-color: #fff3cd;
            border: 1px solid #ffeaa7;
            border-radius: 8px;
            padding: 15px;
            margin: 20px 0;
            color: #856404;
        }
        .footer {
            background-color: #f8f9fa;
            text-align: center;
            padding: 30px 20px;
            color: #6c757d;
            font-size: 14px;
        }
        .security-tips {
            margin-top: 20px;
            font-size: 13px;
            color: #666;
        }
        @media (max-width: 600px) {
            .email-container {
                margin: 20px;
                border-radius: 12px;
            }
            .content {
                padding: 30px 20px;
            }
            .code {
                font-size: 28px;
                letter-spacing: 4px;
            }
        }
    </style>
</head>
<body>
    <div class="email-container">
        <div class="header">
            <div class="logo">🎨 AIICG Wallpaper</div>
            <div class="subtitle">AI Intelligent Wallpaper Generator</div>
        </div>
        
        <div class="content">
            <div class="greeting">Dear %s,</div>
            
            <p>Thank you for registering with AIICG Wallpaper! To complete your registration, please verify your email address using the verification code below:</p>
            
            <div class="code-section">
                <div class="code-label">Your Verification Code</div>
                <div class="code-box">
                    <div class="code">%s</div>
                </div>
            </div>
            
            <div class="instructions">
                <strong>📋 How to use this code:</strong>
                <ol>
                    <li>Return to the AIICG Wallpaper registration page</li>
                    <li>Enter this 6-digit verification code</li>
                    <li>Complete your account setup</li>
                </ol>
            </div>
            
            <div class="warning">
                <strong>⚠️ Important Security Information:</strong>
                <ul>
                    <li>This code is valid for <strong>%d minutes only</strong></li>
                    <li>You have <strong>3 attempts</strong> to enter the correct code</li>
                    <li>Never share this code with anyone</li>
                    <li>If you didn't request this verification, please ignore this email</li>
                </ul>
            </div>
            
            <p>After verification, you'll be able to:</p>
            <ul style="color: #667eea;">
                <li>🎨 Generate unlimited AI wallpapers</li>
                <li>💾 Save and manage your creations</li>
                <li>⭐ Favorite wallpapers you love</li>
                <li>🎭 Access premium features</li>
            </ul>
        </div>
        
        <div class="footer">
            <p>This email was sent automatically, please do not reply.</p>
            <p><strong>© 2024 AIICG Wallpaper</strong> - Let AI Create Beauty for You</p>
            
            <div class="security-tips">
                <p>🔒 <strong>Security Tips:</strong></p>
                <p>• Always verify the sender's email address<br>
                • AIICG will never ask for your password via email<br>
                • Report suspicious emails to security@aiicg.com</p>
            </div>
        </div>
    </div>
</body>
</html>`, username, code, CODE_EXPIRY_MINUTES)
}

// generateVerificationText generate plain text email template
func (e *EmailServiceOptimized) generateVerificationText(code, username string) string {
	return fmt.Sprintf(`
AIICG Wallpaper - Email Verification Code

Dear %s,

Thank you for registering with AIICG Wallpaper!

Your verification code is: %s

This code is valid for %d minutes and you have 3 attempts to enter it correctly.

How to use this code:
1. Return to the AIICG Wallpaper registration page
2. Enter this 6-digit verification code
3. Complete your account setup

Security reminders:
- Never share this code with anyone
- If you didn't request this verification, please ignore this email
- AIICG will never ask for your password via email

After verification, you'll be able to generate unlimited AI wallpapers, save your creations, and access premium features.

This email was sent automatically, please do not reply.
© 2024 AIICG Wallpaper - Let AI Create Beauty for You
`, username, code, CODE_EXPIRY_MINUTES)
}

File: src/lib/email/email-service.ts
/**
 * 邮件服务 - 支持多种邮件提供商
 */

interface EmailConfig {
  provider: 'smtp' | 'resend' | 'sendgrid' | 'mock';
  config: Record<string, any>;
}

interface EmailTemplate {
  subject: string;
  html: string;
  text: string;
}

interface SendEmailOptions {
  to: string;
  subject: string;
  html: string;
  text?: string;
}

class EmailService {
  private config: EmailConfig;

  constructor() {
    // 根据环境变量选择邮件提供商
    this.config = this.getEmailConfig();
  }

  private getEmailConfig(): EmailConfig {
    // 优先级：Resend > SendGrid > SMTP > Mock
    if (process.env.RESEND_API_KEY) {
      console.log('📧 使用Resend邮件服务');
      return {
        provider: 'resend',
        config: {
          apiKey: process.env.RESEND_API_KEY,
          from: process.env.EMAIL_FROM || 'noreply@aiicg.com'
        }
      };
    }

    if (process.env.SENDGRID_API_KEY) {
      return {
        provider: 'sendgrid',
        config: {
          apiKey: process.env.SENDGRID_API_KEY,
          from: process.env.EMAIL_FROM || 'noreply@aiicg.com'
        }
      };
    }

    if (process.env.SMTP_HOST) {
      return {
        provider: 'smtp',
        config: {
          host: process.env.SMTP_HOST,
          port: parseInt(process.env.SMTP_PORT || '587'),
          secure: process.env.SMTP_SECURE === 'true',
          auth: {
            user: process.env.SMTP_USER,
            pass: process.env.SMTP_PASS
          },
          from: process.env.EMAIL_FROM || 'noreply@aiicg.com'
        }
      };
    }

    // 默认使用模拟邮件服务（开发环境）
    console.log('📧 使用Mock邮件服务（开发环境）');
    return {
      provider: 'mock',
      config: {
        from: 'noreply@aiicg.com'
      }
    };
  }

  /**
   * 发送邮件
   */
  async sendEmail(options: SendEmailOptions): Promise<{ success: boolean; error?: string }> {
    try {
      console.log(`📧 发送邮件 [${this.config.provider}]: ${options.to} - ${options.subject}`);

      switch (this.config.provider) {
        case 'resend':
          return await this.sendWithResend(options);
        case 'sendgrid':
          return await this.sendWithSendGrid(options);
        case 'smtp':
          return await this.sendWithSMTP(options);
        case 'mock':
          return await this.sendWithMock(options);
        default:
          throw new Error(`不支持的邮件提供商: ${this.config.provider}`);
      }
    } catch (error) {
      console.error('发送邮件失败:', error);
      return {
        success: false,
        error: error instanceof Error ? error.message : '发送邮件失败'
      };
    }
  }

  /**
   * 使用Resend发送邮件
   */
  private async sendWithResend(options: SendEmailOptions) {
    try {
      console.log(`📧 正在通过Resend发送邮件到: ${options.to}`);

      const response = await fetch('https://api.resend.com/emails', {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${this.config.config.apiKey}`,
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({
          from: this.config.config.from,
          to: [options.to],
          subject: options.subject,
          html: options.html,
          text: options.text
        })
      });

      const responseData = await response.json();

      if (!response.ok) {
        console.error('❌ Resend API错误:', responseData);
        throw new Error(`Resend API错误: ${response.status} - ${JSON.stringify(responseData)}`);
      }

      console.log('✅ 邮件发送成功:', responseData);
      return { success: true };
    } catch (error) {
      console.error('❌ Resend邮件发送失败:', error);
      throw error;
    }
  }

  /**
   * 使用SendGrid发送邮件
   */
  private async sendWithSendGrid(options: SendEmailOptions) {
    const response = await fetch('https://api.sendgrid.com/v3/mail/send', {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${this.config.config.apiKey}`,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        personalizations: [{
          to: [{ email: options.to }]
        }],
        from: { email: this.config.config.from },
        subject: options.subject,
        content: [
          {
            type: 'text/html',
            value: options.html
          }
        ]
      })
    });

    if (!response.ok) {
      const error = await response.text();
      throw new Error(`SendGrid API错误: ${response.status} - ${error}`);
    }

    return { success: true };
  }

  /**
   * 使用SMTP发送邮件
   */
  private async sendWithSMTP(options: SendEmailOptions) {
    // 这里需要使用nodemailer或类似的SMTP客户端
    // 为了简化，暂时返回成功（实际项目中需要实现）
    console.log('📧 SMTP邮件发送（模拟）:', options);
    return { success: true };
  }

  /**
   * 模拟邮件发送（开发环境）
   */
  private async sendWithMock(options: SendEmailOptions) {
    console.log('\n📧 ===== 模拟邮件发送 =====');
    console.log(`📮 收件人: ${options.to}`);
    console.log(`📝 主题: ${options.subject}`);
    console.log('📄 邮件内容:');

    // 从HTML中提取验证码
    const codeMatch = options.html.match(/<div class="code">(\d+)<\/div>/);
    if (codeMatch) {
      console.log(`\n🔑 验证码: ${codeMatch[1]}`);
      console.log(`⏰ 有效期: 5分钟`);
    }

    console.log('\n💡 提示: 这是开发环境的模拟邮件，验证码已在上方显示');
    console.log('========================\n');

    // 模拟网络延迟
    await new Promise(resolve => setTimeout(resolve, 500));

    return { success: true };
  }

  /**
   * 生成验证码邮件模板
   */
  generateVerificationEmail(code: string, username: string): EmailTemplate {
    const subject = 'AIICG壁纸站 - 邮箱验证码';

    const html = `
      <!DOCTYPE html>
      <html lang="zh-CN">
      <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>邮箱验证码</title>
        <style>
          body {
            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, 'Helvetica Neue', Arial, sans-serif;
            line-height: 1.6;
            color: #333;
            margin: 0;
            padding: 0;
            background-color: #f8fafc;
          }
          .container {
            max-width: 600px;
            margin: 0 auto;
            padding: 20px;
            background-color: #ffffff;
            border-radius: 12px;
            box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
          }
          .header {
            text-align: center;
            padding: 30px 0;
            border-bottom: 2px solid #4F46E5;
            margin-bottom: 30px;
          }
          .logo {
            font-size: 28px;
            font-weight: bold;
            color: #4F46E5;
            margin-bottom: 8px;
          }
          .subtitle {
            color: #6B7280;
            font-size: 16px;
          }
          .content {
            padding: 0 20px;
          }
          .greeting {
            font-size: 18px;
            margin-bottom: 20px;
            color: #1F2937;
          }
          .code-box {
            background: linear-gradient(135deg, #667eea 0%, #764ba2 100%);
            border-radius: 12px;
            padding: 30px;
            text-align: center;
            margin: 30px 0;
            box-shadow: 0 8px 25px rgba(102, 126, 234, 0.3);
          }
          .code {
            font-size: 36px;
            font-weight: bold;
            color: #ffffff;
            letter-spacing: 6px;
            font-family: 'Courier New', monospace;
            text-shadow: 0 2px 4px rgba(0, 0, 0, 0.3);
          }
          .code-label {
            color: #E5E7EB;
            font-size: 14px;
            margin-bottom: 10px;
            text-transform: uppercase;
            letter-spacing: 1px;
          }
          .warning {
            background-color: #FEF3C7;
            border: 1px solid #F59E0B;
            border-radius: 8px;
            padding: 15px;
            margin: 20px 0;
            color: #92400E;
          }
          .footer {
            text-align: center;
            padding: 30px 0 20px;
            border-top: 1px solid #E5E7EB;
            color: #6B7280;
            font-size: 14px;
            margin-top: 40px;
          }
          .footer p {
            margin: 5px 0;
          }
          @media only screen and (max-width: 600px) {
            .container {
              margin: 10px;
              padding: 15px;
            }
            .code {
              font-size: 28px;
              letter-spacing: 4px;
            }
          }
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <div class="logo">🎨 AIICG壁纸站</div>
            <div class="subtitle">AI智能壁纸生成器</div>
          </div>

          <div class="content">
            <div class="greeting">亲爱的 ${username}，</div>
            <p>感谢您注册AIICG壁纸站！为了确保账户安全，请使用以下验证码完成邮箱验证：</p>

            <div class="code-box">
              <div class="code-label">验证码</div>
              <div class="code">${code}</div>
            </div>

            <div class="warning">
              <strong>⚠️ 重要提醒：</strong>
              <ul style="margin: 10px 0; padding-left: 20px;">
                <li>验证码有效期：<strong>5分钟</strong></li>
                <li>请勿将验证码告诉他人</li>
                <li>如果您没有注册AIICG壁纸站，请忽略此邮件</li>
              </ul>
            </div>

            <p>完成验证后，您将可以：</p>
            <ul style="color: #4F46E5; padding-left: 20px;">
              <li>🎨 无限制AI壁纸生成</li>
              <li>💾 保存和管理您的作品</li>
              <li>⭐ 收藏喜欢的壁纸</li>
              <li>🎁 获得精美随机头像</li>
            </ul>
          </div>

          <div class="footer">
            <p>此邮件由系统自动发送，请勿回复</p>
            <p>© 2024 AIICG壁纸站 - 让AI为您创造美丽</p>
          </div>
        </div>
      </body>
      </html>
    `;

    const text = `
AIICG壁纸站 - 邮箱验证码

亲爱的 ${username}，

感谢您注册AIICG壁纸站！请使用以下验证码完成邮箱验证：

验证码：${code}

验证码有效期：5分钟

如果您没有注册AIICG壁纸站，请忽略此邮件。

此邮件由系统自动发送，请勿回复
© 2024 AIICG壁纸站 - AI智能壁纸生成器
    `;

    return { subject, html, text };
  }

  /**
   * 检查邮件服务是否可用
   */
  async isAvailable(): Promise<boolean> {
    return this.config.provider !== 'mock' || process.env.NODE_ENV === 'development';
  }

  /**
   * 获取当前邮件提供商
   */
  getProvider(): string {
    return this.config.provider;
  }
}

// 导出单例
export const emailService = new EmailService();
